util/execdetails/execdetails.go

```diff
@@ -189,17 +187,46 @@ type CopDetails struct {
 	RocksdbBlockReadCount uint64
 	// RocksdbBlockReadByte is the total number of bytes from block reads.
 	RocksdbBlockReadByte uint64
+	// WaitWallTimeMs is the off-cpu wall time which is elapsed in TiKV side. Usually this includes queue waiting time and
+	// other kind of waitings in series.
+	ProcessTime time.Duration
+	// Off-cpu and on-cpu wall time elapsed to actually process the request payload. It does not
+	// include `wait_wall_time`.
+	// This field is very close to the CPU time in most cases. Some wait time spend in RocksDB
+	// cannot be excluded for now, like Mutex wait time, which is included in this field, so that
+	// this field is called wall time instead of CPU time.
+	WaitTime time.Duration
 }
 
 // Merge merges lock keys execution details into self.
-func (cd *CopDetails) Merge(copDetails *CopDetails) {
-	cd.TotalKeys += copDetails.TotalKeys
-	cd.ProcessedKeys += copDetails.ProcessedKeys
-	cd.RocksdbDeleteSkippedCount += copDetails.RocksdbDeleteSkippedCount
-	cd.RocksdbKeySkippedCount += copDetails.RocksdbKeySkippedCount
-	cd.RocksdbBlockCacheHitCount += copDetails.RocksdbBlockCacheHitCount
-	cd.RocksdbBlockReadCount += copDetails.RocksdbBlockReadCount
-	cd.RocksdbBlockReadByte += copDetails.RocksdbBlockReadByte
+func (sd *ScanDetail) Merge(scanDetail *ScanDetail) {
+	sd.TotalKeys += scanDetail.TotalKeys
+	sd.ProcessedKeys += scanDetail.ProcessedKeys
+	sd.RocksdbDeleteSkippedCount += scanDetail.RocksdbDeleteSkippedCount
+	sd.RocksdbKeySkippedCount += scanDetail.RocksdbKeySkippedCount
+	sd.RocksdbBlockCacheHitCount += scanDetail.RocksdbBlockCacheHitCount
+	sd.RocksdbBlockReadCount += scanDetail.RocksdbBlockReadCount
+	sd.RocksdbBlockReadByte += scanDetail.RocksdbBlockReadByte
+	sd.ProcessTime += scanDetail.ProcessTime
+	sd.WaitTime += scanDetail.WaitTime
+}
+
+func (sd *ScanDetail) String() string {
+	buf := bytes.NewBuffer(make([]byte, 0, 16))
+	if sd != nil {
+		buf.WriteString(ProcessTimeStr + ":" + strconv.FormatFloat(sd.ProcessTime.Seconds(), 'f', -1, 64))
+		buf.WriteString(", " + WaitTimeStr + ":" + strconv.FormatFloat(sd.ProcessTime.Seconds(), 'f', -1, 64))
+		buf.WriteString(", rocksdb{")
+		buf.WriteString(ProcessKeysStr + ": " + strconv.FormatInt(sd.ProcessedKeys, 10))
+		buf.WriteString(", " + TotalKeysStr + ": " + strconv.FormatInt(sd.TotalKeys, 10))
+		buf.WriteString(", " + RocksdbDeleteSkippedCountStr + ": " + strconv.FormatUint(sd.RocksdbDeleteSkippedCount, 10))
```

Comment:
Considering that we have a `rocksdb{...}` at outside, maybe there is no need to have `Rocksdb_xxx` prefix for each fields inside?